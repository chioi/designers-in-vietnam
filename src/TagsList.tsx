import React from "react";
import Tag, {ITag} from "./Tag";

interface ITagsListProps {
  tags: ITag[];
}

const TagsList = (props: ITagsListProps) => {
  const tagWrappers = props.tags.map(tag => <Tag key={tag.id} tag={tag} />);
  return <ul className="horizontal-list">{tagWrappers}</ul>;
};

export default TagsList;
