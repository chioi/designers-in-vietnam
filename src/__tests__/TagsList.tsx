import React from "react";
import { create, ReactTestRenderer } from "react-test-renderer";
import TagsList, { ITag } from "../TagsList";

const tags: ITag[] = [
  {
    id: "jjs234l",
    name: "UX"
  }
];

describe("gets created", () => {
  describe("without tags", () => {
    let component: ReactTestRenderer;

    beforeAll(() => {
      component = create(<TagsList tags={[]} />);
    });

    it("has no list elements", () => {
      expect(component.root.findAllByType("li")).toHaveLength(0);
    });

    it("matches the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });

  describe("with tags", () => {
    let component: ReactTestRenderer;

    beforeAll(() => {
      component = create(<TagsList tags={tags} />);
    });

    it("has no list elements", () => {
      expect(component.root.findAllByType("li")).toHaveLength(tags.length);
    });

    it("matches the snapshot", () => {
      expect(component).toMatchSnapshot();
    });
  });
});
