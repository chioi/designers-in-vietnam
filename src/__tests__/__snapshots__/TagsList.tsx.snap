// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`gets created with tags matches the snapshot 1`] = `
<ul
  className="horizontal-list"
>
  <li
    className="tag"
  >
    <label
      className="tag-content"
      htmlFor="UX-input"
      id="UX-tag"
      onClick={[Function]}
    >
      <input
        id="[object Object]-input"
        type="checkbox"
      />
      UX
    </label>
  </li>
</ul>
`;

exports[`gets created without tags matches the snapshot 1`] = `
<ul
  className="horizontal-list"
/>
`;
